### ISSUE TITLE
[BUG] Token HUD not shown, `game.tokenActionHud is undefined` in console

### ISSUE THREAD

**Issue by hamishcampbell** [Larkinabout/fvtt-token-action-hud-core#143]
**Describe the bug**

Token HUD not visible.

**Steps to reproduce**
1. Install token HUD core & 5e
2. Enable for game world
3. Nothing happens

Pressing H causes the following in console (Firefox):

```
Uncaught TypeError: game.tokenActionHud is undefined
    onDown token-action-hud-core.min.js:1759
    _executeKeybind foundry.js:2500
    _processKeyboardContext foundry.js:2541
    _handleKeyboardEvent foundry.js:2599
    _activateListeners foundry.js:2225
    _activateListeners foundry.js:2225
    initializeKeyboard foundry.js:8850
    setupGame foundry.js:8659
    _initializeGameView foundry.js:9925
    _initializeView foundry.js:9901
    initialize foundry.js:8588
    async* foundry.js:90285
    async* foundry.js:90266
```

Interestingly, if I break there are run `game.initialize()` the HUD now appears and begins working, although with some hooks (e.g. show/hide on `H`) running twice. So it sounds like it's not booting correctly, but some hooks are registered but the instance isn't attached to `game` correctly.

**Versions:**
- Token Action HUD Core: 1.4.8
- Token Action HUD System: token-action-hud-dnd5e 1.4.4
- System: dnd5e 2.2.1
- Foundry VTT: 11.300.0
- Browser: FF 113 & Brave 1.52.117

**Comment by hamishcampbell** [comment]
Running `Hooks.callAll('tokenActionHudCoreReady')` in console appears to cleanly boot it.

**Comment by Larkinabout** [comment]
I've noticed this when Foundry doesn't load straight into a scene. It's because the hooks are nested and the HUD is expecting a canvas. Is this true for your world?

**Comment by hamishcampbell** [comment]
I'm brand-new to Foundry hooks, but looking through the code - it seems that:

```
1. The 'ready' hook triggers `tokenActionHudCoreApiReady` (https://github.com/Larkinabout/fvtt-token-action-hud-core/blob/c27f1b77f9d91c1628510cb54b79aa13647eb7bc/scripts/init.js#L36)
2. this triggers the specific module, which triggers `tokenActionHudSystemReady` https://github.com/Larkinabout/fvtt-token-action-hud-dnd5e/blob/64eae03322c3da765149c441dc9a41c9681e84c1/scripts/init.js#L10
3. which is supposed to trigger `tokenActionHudCoreReady` https://github.com/Larkinabout/fvtt-token-action-hud-core/blob/c27f1b77f9d91c1628510cb54b79aa13647eb7bc/scripts/init.js#L72

However `tokenActionHudCoreReady` isn't registered until `canvasReady` is run, and this appears to happen _after_ the above - or at least it is in my configuration.

It seems to me that `tokenActionHudCoreReady` should be registered _outside_ of `canvasReady` (
https://github.com/Larkinabout/fvtt-token-action-hud-core/blob/c27f1b77f9d91c1628510cb54b79aa13647eb7bc/scripts/init.js#L76). Currently, it doesn't exist at the time the module triggers `tokenActionHudCoreReady` so nothing happens and I don't quite understand why it should happen on every `canvasReady`. E.g. `canvasReady` is called on each scene load, so the hook is presumbly registered multiple times over the lifetime of a session, which doesn't seem right?

**Comment by Larkinabout** [comment]
Good to hear. I'll look at using a different sequence of hooks. It was nested that way to avoid triggering the code before Foundry VTT was ready, but it has this unintended consequence.

**Comment by hamishcampbell** [comment]
Thanks. **tldr;** for future visitors: if the HUD doesn't appear to load correctly, make sure you have an initial scene set.

**Comment by Larkinabout** [comment]
Fixed in Token Action HUD Core 1.4.9.
